bullet_train/lib/bullet_train/core_ext/callbacks_instrumentation.rb

```diff
@@ -0,0 +1,60 @@
+require_relative "callbacks_instrumentation/backport"
+
+module ActiveSupport::Callbacks
+  ClassMethods.prepend Module.new {
+    def set_callback(name, *filter_list, &block)
+      *, options = normalize_callback_params(filter_list.dup, block)
+      Instrumenter.options = options.merge(caller_locations:)
+      super
+    ensure
+      Instrumenter.options = nil
+    end
+  }
+
+  Callback.prepend Module.new {
+    def compiled
+      @compiled = super.then { Instrumenter.build self, _1 } unless defined?(@compiled)
+      @compiled
+    end
+  }
+
+  class Instrumenter < Data.define(:callback, :callable, :options)
+    singleton_class.attr_accessor :options
+
+    def self.build(callback, callable)
+      options ? new(callback, callable, options) : callable
+    end
+
+    def initialize(callback:, callable:, options:)
+      caller_locations = options.delete(:caller_locations)
```

Comment:
I got this error when trying to migrate the database on a fresh new application:

```
Loading application environment...
Loading code in search of Active Record models...
bin/rails aborted!
NoMethodError: undefined method `first' for nil (NoMethodError)

      @full_name = (callback.name == :commit) ? caller_locations.first.base_label : "#{callback.kind}_#{callback.name}"
                                                                ^^^^^^
/home/gazayas/work/bt/bullet_train/local/bullet_train-core/bullet_train/lib/bullet_train/core_ext/callbacks_instrumentation.rb:31:in `initialize'
```

Since `caller_locations` is `nil` here, I wonder if we should simply `return` the method in that case? I can't say I know too much about this Instrumenter class to say that that's the right decision.

Reply:
As a side note, the `Loading code in search of Active Record models` comes from `rails-erd`. Dropping this for reference because we do some eager loading here: https://github.com/bullet-train-co/bullet_train/pull/1200